### PR TITLE
Add bindings for the Ristretto255 API

### PR DESCRIPTION
**Pull request by eploko**
This is some preliminary effort to expose the Ristretto255 API in libsodium.

Some key points:
* Test vectors fetched from the [Ristretto Group's test vectors page](https://ristretto.group/test_vectors/ristretto255.html).
* All Ristretto255-related tests ported from libsodium.
* An additional hash-to-group test ported from the aforementioned [Ristretto Group's test vectors page](https://ristretto.group/test_vectors/ristretto255.html).
* The `crypto_core_ristretto255_from_hash ` fn from libsodium has been exposed as `hash->group` on the `caesium.crypto.core.ristretto255` ns.
* Also `is_valid_point` and `is_canonical` are exposed as `valid-point?` and `canonical?` in the corresponding namespases.

I guess this is now ready for review. Please let me know if anything needs to be changed or improved for this PR to be accepted and merged.

### CODE REVIEW

File: dev-resources/vectors/README.md [hunk 1]
```diff
@@ -1 +1,15 @@
-These vectors were taken from Kalium's test suite.
+The vectors in the following folders were taken from Kalium's test suite:
```
Comment (lvh) [inline dev-resources/vectors/README.md:1]: Same comment as on the other PR; please merge master in, delete this README, add a README into the specific directory for the test vector you're adding.

**Comment by lvh** [comment]
Please consider adding a CHANGELOG entry.

**Comment by lvh** [comment]
This looks great modulo above administrative tweaks. Thank you for your contribution!

File: dev-resources/vectors/README.md (hunk 1 above)
Comment (eploko) [reply dev-resources/vectors/README.md]: Done.

**Comment by eploko** [comment]
Merged the `master` branch in, added the separate `README.md` file in the `dev-resources/vectors/ristretto255` folder, and added a CHANGELOG entry.

**Comment by lvh** [comment]
Looks great, thanks!

**Pull request merged**